### PR TITLE
refactor(config): document and validate ReviewConfig::min_budget_for_call_graph coupling

### LINKED ISSUE #1362: refactor(config): simplify ReviewConfig -- remove or replace min_budget_for_call_graph
## Context

`ReviewConfig` has 9 fields controlling prompt assembly for PR review. Eight are straightforward size caps on independent components (full file content, diff/patch, instructions, dep enrichments). The ninth is different:

```rust
// review_context.rs
let budget_remaining = max_prompt_chars.saturating_sub(estimated_size);
let should_enable_cg = deep || budget_remaining > config.min_budget_for_call_graph;
```

`min_budget_for_call_graph` (default: 20,000) is a threshold on *remaining budget after estimation*, not a cap on component size. It leaks a budget-accounting implementation detail into user-facing configuration.

## Problem

- The field is the only one in `ReviewConfig` that requires the user to reason about *internal prompt arithmetic* (estimate size, subtract from ceiling, compare remainder) to tune correctly. All other fields are independently meaningful.
- `estimated_size` is computed before call graph fetch, so the gate fires on a pre-call-graph estimate. The 20,000 char headroom figure is not self-documenting and is hard to set correctly without understanding the estimation logic.
- The `deep` flag already provides an explicit opt-in override. The budget gate is a secondary heuristic on top of it.

## Proposed investigation

Evaluate whether `min_budget_for_call_graph` can be eliminated or replaced:

1. **Remove it entirely** -- always build the call graph (it is text; the existing `max_prompt_chars` ceiling already trims the final prompt). The gate was defensive, but if the prompt builder already enforces the budget ceiling, the pre-gate may be redundant.
2. **Replace with a ratio** -- express as a fraction of `max_prompt_chars` (e.g. 1/6) so it scales automatically when the ceiling is changed, removing the coupling.
3. **Keep as-is but document the arithmetic** -- lowest value option; documents the dependency on `max_prompt_chars` explicitly in the field doc.

Also audit the remaining 8 fields for similar issues:
- Are `max_diff_chars` and `max_patch_chars_per_file` both needed, or does one subsume the other in practice?
- Are defaults consistent with each other (e.g. `max_chars_per_file * max_full_content_files` vs. `max_prompt_chars`)?

## Acceptance criteria

- [ ] `min_budget_for_call_graph` is either removed, replaced with a self-contained expression, or its coupling to `max_prompt_chars` is made explicit and tested.
- [ ] Any removed config field is handled with a deprecation path (env var ignored with a warning, `config.toml` key ignored gracefully).
- [ ] No behaviour change for users who rely on the current defaults.
- [ ] Field count in `ReviewConfig` does not increase.

### PR DESCRIPTION
**Pull request by clouatre**
## Summary

Documents and validates the internal coupling between `ReviewConfig::min_budget_for_call_graph` and `max_prompt_chars`. The call graph is built only when `budget_remaining > min_budget_for_call_graph` (where `budget_remaining = max_prompt_chars - estimated_size`). A misconfigured threshold can silently disable call graph enrichment or make it apply only for the largest diffs.

Changes:
- Added `ReviewConfig::validate_consistency()` returning `Vec<String>` warnings for two misconfiguration patterns: threshold >= ceiling (gate permanently off) and threshold > ceiling/2 (gate rarely fires).
- Improved doc comment on `min_budget_for_call_graph` to explain the arithmetic coupling and strict `>` comparison explicitly.
- Called `validate_consistency()` at config load time inside `FileConfigSource::load`, alongside the existing `cache.validate()` call. Warnings fire once at startup when the config is parsed, not on every review.
- Added 6 unit tests: default config (happy path), threshold equals max, threshold over half, two `should_enable_call_graph` strict-boundary cases, and `deep=true` bypasses the budget gate entirely.
- Updated `docs/CONFIGURATION.md`, `docs/GITHUB_ACTION.md`, and `action.yml` (both the `deep` input description and `min-budget-for-call-graph` input description) with the explicit arithmetic coupling note.

No behavioral changes; no new struct fields added.

## Files Changed

- `crates/aptu-core/src/config/review.rs` — `validate_consistency()`, improved doc comment, 4 tests
- `crates/aptu-core/src/config/loader.rs` — `validate_consistency()` call at config load time
- `crates/aptu-core/src/ai/review_context.rs` — 2 tests (`should_enable_call_graph` boundary + `deep` override)
- `action.yml` — coupling note in `deep` and `min-budget-for-call-graph` input descriptions
- `docs/CONFIGURATION.md` — coupling explanation
- `docs/GITHUB_ACTION.md` — coupling documentation

## Test Plan

- 6 new unit tests pass; cargo clippy clean; cargo fmt clean.

Closes #1362

### CODE REVIEW

File: crates/aptu-core/src/ai/review_context.rs [hunk 1]
```diff
@@ -185,6 +186,18 @@ pub async fn build_review_context(
     deep: bool,
     review_config: &ReviewConfig,
 ) -> crate::Result<ReviewContext> {
+    // Validate configuration consistency at build entry.
+    // OnceLock ensures each distinct warning is emitted at most once per process,
```
[the remark this answers is not in the document]
Comment (clouatre) [reply crates/aptu-core/src/ai/review_context.rs]: Noted. The current design calls `validate_consistency()` inside the `OnceLock::get_or_init` closure, so the closure captures the config by reference at first call. If config reloading is added in the future the `OnceLock` would need to be keyed per-config snapshot (or replaced with an `AtomicBool` per warning). Added a FIXME comment in the source to flag this for that future change.

File: crates/aptu-core/src/ai/review_context.rs [hunk 2]
```diff
@@ -185,6 +186,21 @@ pub async fn build_review_context(
...
+    // preventing log spam on every PR review when the config is misconfigured.
+    // FIXME: if per-session config reloading is added, replace OnceLock<()> with
+    // an AtomicBool per warning (or key on a config snapshot hash) so that changed
+    // values are re-validated after reload.
```
[the remark this answers is not in the document]
Comment (clouatre) [reply crates/aptu-core/src/ai/review_context.rs]: Moved validation to config load time (inside the `FileConfigSource::load` post-deserialize block in `loader.rs`, alongside the existing `cache.validate()` call). The `OnceLock` and the entire validation block have been removed from `build_review_context`. Warnings now fire once at startup when the config is parsed, not on every review.
